### ISSUE TITLE
[FEATURE REQUEST] Disable sea salt aerosol debromination and include as an option instead

### ISSUE THREAD

**Issue by msulprizio** [geoschem/geos-chem#1145]
The GEOS-Chem Steering Committee has asked that we disable sea salt aerosol debromination and include it as an option instead. This came out of concern over the [GEOS-Chem 13.4.0 benchmark](http://wiki.seas.harvard.edu/geos-chem/index.php/GEOS-Chem_13.4.0#1-year_fullchem_GEOS-Chem_Classic_version_comparison_.2813.4.0-rc.1_vs_13.3.0-rc.0.29) where ozone concentrations are too low, especially over the Southern Ocean. See the [GCSC meeting minutes](https://geos-chem.seas.harvard.edu/files/acmg-geos/files/gcsc_minutes_23feb2022_01.pdf) for more details. Essentially, we need to zero out the necessary reactions and rerun the 13.4.0 benchmark simulations.

**Comment by djxjacob** [comment]
Hi Melissa – Mike is also making a  fix to the SO2+HOBR reaction in KPP that should go into 13.4.0.  Daniel

From: Melissa Sulprizio ***@***.***>
Sent: Tuesday, March 1, 2022 11:36 AM
To: geoschem/geos-chem ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [geoschem/geos-chem] [FEATURE REQUEST] Disable sea salt aerosol debromination and include as an option instead (Issue #1145)


The GEOS-Chem Steering Committee has asked that we disable sea salt aerosol debromination and include it as an option instead. This came out of concern over the GEOS-Chem 13.4.0 benchmark<http://wiki.seas.harvard.edu/geos-chem/index.php/GEOS-Chem_13.4.0#1-year_fullchem_GEOS-Chem_Classic_version_comparison_.2813.4.0-rc.1_vs_13.3.0-rc.0.29> where ozone concentrations are too low, especially over the Southern Ocean. See the GCSC meeting minutes<https://geos-chem.seas.harvard.edu/files/acmg-geos/files/gcsc_minutes_23feb2022_01.pdf> for more details. Essentially, we need to zero out the necessary reactions and rerun the 13.4.0 benchmark simulations.

—
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_geoschem_geos-2Dchem_issues_1145&d=DwMCaQ&c=WO-RGvefibhHBZq3fL85hQ&r=rhdjeCmvErplnjGEj5OoH_CDkPdqB-BDUxqwWC1AzIs&m=dYkK_NlBFovklL11XOUAEEwCNiudqwGvG3rbV4ExX6Sir-C_wcm_r2IM72Z_RekH&s=vw_vlB0C-qXPpoopmu4383xpgXXmH8T3XVMdIxg_kfY&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AKHHSVK5UG7OZREE7I5HJ4LU5ZBPFANCNFSM5PURMUGA&d=DwMCaQ&c=WO-RGvefibhHBZq3fL85hQ&r=rhdjeCmvErplnjGEj5OoH_CDkPdqB-BDUxqwWC1AzIs&m=dYkK_NlBFovklL11XOUAEEwCNiudqwGvG3rbV4ExX6Sir-C_wcm_r2IM72Z_RekH&s=1CdUuCQcZjqAX5YHGnWf6wJWarO_eGhkXfA7aXI6ktY&e=>.
Triage notifications on the go with GitHub Mobile for iOS<https://urldefense.proofpoint.com/v2/url?u=https-3A__apps.apple.com_app_apple-2Dstore_id1477376905-3Fct-3Dnotification-2Demail-26mt-3D8-26pt-3D524675&d=DwMCaQ&c=WO-RGvefibhHBZq3fL85hQ&r=rhdjeCmvErplnjGEj5OoH_CDkPdqB-BDUxqwWC1AzIs&m=dYkK_NlBFovklL11XOUAEEwCNiudqwGvG3rbV4ExX6Sir-C_wcm_r2IM72Z_RekH&s=13UGbMp_aJx2ZeuL4c54eC7z3w9jTqz41tHli-hLXEM&e=> or Android<https://urldefense.proofpoint.com/v2/url?u=https-3A__play.google.com_store_apps_details-3Fid-3Dcom.github.android-26referrer-3Dutm-5Fcampaign-253Dnotification-2Demail-2526utm-5Fmedium-253Demail-2526utm-5Fsource-253Dgithub&d=DwMCaQ&c=WO-RGvefibhHBZq3fL85hQ&r=rhdjeCmvErplnjGEj5OoH_CDkPdqB-BDUxqwWC1AzIs&m=dYkK_NlBFovklL11XOUAEEwCNiudqwGvG3rbV4ExX6Sir-C_wcm_r2IM72Z_RekH&s=z4buSRTqUK0PMu2axaABF-1zUIlFC-GxUJBHm1HWI_U&e=>.

**Comment by msulprizio** [comment]
Thanks Daniel. We will be sure to include that fix as well.

**Comment by msulprizio** [comment]
Also tagging @viral211, @tsherwen, @msl3v here for any guidance or input

**Comment by yantosca** [comment]
@viral211 wrote:

Here are the reactions involved in sea salt aerosol debromination:
```console
ClNO3 + BrSALA = BrCl + HNO3 : ClNO3uptkByBrSALA( State_Het );
ClNO3 + BrSALC = BrCl + HNO3 : ClNO3uptkByBrSALC( State_Het ); 
ClNO2 + BrSALA = BrCl + HNO2 : ClNO2uptkByBrSALA( State_Het ); 
ClNO2 + BrSALC = BrCl + HNO2 : ClNO2uptkByBrSALC( State_Het ); 
HOBr + BrSALA = Br2 :          HOBrUptkByBrSALA( State_Het );
HOBr + BrSALC = Br2 :          HOBrUptkByBrSALC( State_Het );
O3 + BrSALA = HOBr :           O3uptkByBrSALA( State_Het );
O3 + BrSALC = HOBr :           O3uptkByBrSALC( State_Het );
IONO + BrSALA = IBr + HNO2 :   IbrkdnByAcidBrSALA( SR_MW(ind_IONO), C(ind_IONO), 0.02_dp, State_Het ); 
IONO + BrSALC = IBr + HNO2 :   IbrkdnByAcidBrSALC( SR_MW(ind_IONO), C(ind_IONO), 0.02_dp, State_Het );
IONO2 + BrSALA = IBr + HNO3 :  IbrkdnByAcidBrSALA( SR_MW(ind_IONO2), C(ind_IONO2), 0.01_dp, State_Het );
IONO2 + BrSALC = IBr + HNO3 :  IbrkdnByAcidBrSALC( SR_MW(ind_IONO2), C(ind_IONO2), 0.01_dp, State_Het );
```

**Comment by yantosca** [comment]
I think the easiest way to proceed toggling SSA debromination on/off is to add a logical field to the HetState type in `KPP/fullchem/fullchem.kpp`:
```F90
  TYPE, PUBLIC :: HetState
     REAL(dp) :: AVO            ! Avogadro's constant [molec/mol]
     LOGICAL  :: debugBox       ! Are we in a debugging box?
     LOGICAL  :: natSurface     ! Is there NAT in this box?
     LOGICAL  :: pscBox         ! Are there polar strat clouds?
     LOGICAL  :: stratBox       ! Are we in the stratosphere
     LOGICAL  :: ssaDebrom      ! Toggle SSA debromination on/off
    ... etc ...
```
and then to populate that with either `.TRUE.` or `.FALSE.` in `KPP/fullchem/fullchem_HetStateFuncs.F90`, in routine `fullchem_SetState_Het`:
```F90
    ! ... check if there is surface NAT at this grid box
    H%natSurface = ( H%pscBox .and. ( C(ind_NIT) > 0.0_dp )                 )

    !========================================================================
    ! Toggle sea salt aerosol debromination on or off
    !========================================================================
    H%ssaDebrom = .FALSE.
``` 

Then in each of the rate-law functions mentioned by @viral211 above (these are in `KPP/fullchem/fullchem_rateLawFuncs.F90`), you can put a shunt at the top of the function, such as:
```F90
  FUNCTION ClNO3uptkByBrSALA( H ) RESULT( k )
    !
    ! Computes rxn rate [1/s] of ClNO3 + BrSALA.
    !
    TYPE(HetState), INTENT(IN) :: H              ! Hetchem State
    REAL(dp)                   :: k              ! Rxn rate [1/s]
    !
    REAL(dp) :: area, branch, branchBr, gamma, srMw
    !
    k    = 0.0_dp
    IF ( .not. H%ssaDebrom ) RETURN    ! Exit if SSA debromination is false
    srMw = SR_MW(ind_ClNO3)
    !
    ... etc ...   
  END FUNCTION ClNO3uptkByBrSALA
```

**Comment by viral211** [comment]
There were a couple of reactions missing in the above list:
HOI + BrSALA = IBr :           IbrkdnByAcidBrSALA( SR_MW(ind_HOI), C(ind_HOI), 0.01_dp, State_Het );
HOI + BrSALC = IBr :           IbrkdnByAcidBrSALC( SR_MW(ind_HOI), C(ind_HOI), 0.01_dp, State_Het );

**Comment by yantosca** [comment]
The other thing you could do is to manually set the rate constants for those reactions to `0.0_dp` (and comment out the function calls) in `KPP/fullchem/fullchem.eqn`, but then if someone wanted to turn on SSA debromination, you would have to restore the function calls and rebuild KPP.  The shunting would take more CPU cycles but it would be easier for users as you wouldn't need to rebuild the mechanism each time you toggle SSA debromination on or off.

**Comment by msulprizio** [comment]
@viral211, Xuan Wang, and @yantosca Thanks for your feedback! I am working on these code updates now and will let you know when a pull request is ready.

**Comment by msulprizio** [comment]
The code modifications for turning off SSA debromination are now in pull request #1155. The plots and tables of a 1-month benchmark isolating the impact from this update can be found at: http://ftp.as.harvard.edu/gcgrid/geos-chem/validation/GCC_13.4.0-rc.2+SSAdebromOff/BenchmarkResults/.

In particular, here is the impact on surface O3:
<img width="524" alt="Screen Shot 2022-03-09 at 8 47 33 AM" src="https://user-images.githubusercontent.com/4805437/157454361-cdc05024-2738-4fb7-945e-18dc854960aa.png">

**Comment by djxjacob** [comment]
Hi Melissa –

That simulation totally kills tropospheric bromine, which is bizarre because in the free troposphere the source from the organobromines particularly CHBr3 should still be important.  Quoting Wang2021, “Bromocarbon gases contribute only 0.54 GgBra-1 to Bry but still dominate the Bry source in the free troposphere.” I’m wondering what’s going on. Even tropospheric HBr is killed which makes me suspect that the problem is not in the hetchem but in some zeroing of Bry species.

Daniel

**Comment by viral211** [comment]
This may be happening because we still allow the uptake of HBr to BrSALA/C. So when we zero out BrSALA/C at each time step, we are just losing Bry. We could disable this reaction when we disable sea salt debromination (zero out the rate). Or, instead of zeroing out BrSALA/C in the C array, we could zero it out in way that only RateLawFuncs "sees" it as such, and we still conserve Bry. Sorry, didn't think of this before.

**Comment by msulprizio** [comment]
@viral211 wrote:

>A better way to disable debromination is to disable emissions of BrSALA/C instead of disabling the heterogeneous reactions that recycle Bry and are important for maintaining Bry (from CHBrx) in the free troposphere. 
>
>I think we can do this simply by setting  Model sea salt Br- to False in HEMCO_Config.rc. But I am not sure how you'd tie this to the new input.geos flag that turns sea salt debromination on/off. 

I am working on implementing and testing this now.

**Comment by msulprizio** [comment]
Pull request #1177 now turns off SSA debromination using switches in HEMCO_Config.rc. I didn't think there was a need to include a switch in input.geos too because this update does not impact the GEOS-Chem source code. Including a switch in both input.geos and HEMCO_Config.rc would be redundant and may cause user confusion.

I ran a 1-month simulation to evaluate the impact of turning off SSA debromination by disabling BrSALA and BrSALC emissions. The results may be viewed at http://ftp.as.harvard.edu/gcgrid/geos-chem/validation/GCC_13.4.0-rc.2+SSAdebromOff_viaHEMCO/BenchmarkResults/. 

Here is the impact on surface ozone and Bry:

![Screen Shot 2022-03-17 at 6 00 46 PM](https://user-images.githubusercontent.com/4805437/158901730-78d5f138-f2c7-4b66-b517-cb689fcc3f90.png)

![Screen Shot 2022-03-17 at 6 00 20 PM](https://user-images.githubusercontent.com/4805437/158901743-8510dce0-66b3-46f7-82db-89d2fb8eda5f.png)

**Comment by msulprizio** [comment]
@viral211 wrote:

>I looked at the Br tropospheric mass, and we are now back to version 13.0 levels (~2 Gg of BrOx and ~20 Gg of Bry). We can expect to see ozone jump back to 13.0 levels in the 1-yr benchmark.

This update has now been merged into `dev` and will be included in the 13.4.0. The official 1-month and 1-year benchmarks for that version will be resubmitted soon.
